pdr_backend/lake/gql_data_factory.py

```diff
@@ -137,18 +137,49 @@ def _update(self, fin_ut: int):
                 continue
 
             # to satisfy mypy, get an explicit function pointer
-            do_fetch: Callable[[str, int, int, Dict], pl.DataFrame] = record["fetch_fn"]
-
-            # call the function
-            print(f"    Fetching {k}")
-            df = do_fetch(self.ppss.web3_pp.network, st_ut, fin_ut, record["config"])
-
-            # postcondition
-            if len(df) > 0:
-                assert df.schema == record["schema"]
-
-                # save to parquet
-                self._save_parquet(filename, df)
+            do_fetch: Callable[[str, int, int, int, int, Dict], pl.DataFrame] = record[
+                "fetch_fn"
+            ]
```

Comment:
Fantastic, now all subgraph fetches handle: start, end, etc... but have no loop.

do_fetch/subgraph logic used to handle the pagination + subgraph loop, but now that's been moved to the core _update() loop of GQLDataFactory... but this is dirtying things up.

Please move the pagination/fetching loop + save-backoff logic into their own functions and parameterize them.

The _update() function for gql_data_factory() should be primarily to loop/kickoff all table updates.... 

Please do so by breaking up this function into 2 parts:
```
1. Handling do_fetch/pagination loop
2. Save-Backoff logic
```

While doing this, please adjust the param names as well so they are more objective. When you put these in their respective function, you may choose to adjust the names.
```
save_to_file_offset + fetched_since_last_save => save_backoff_limit, save_backoff_count 
records_per_page, offset => pagination_limit, pagination_offset
```

Nearly there!

Please get @kdetry or @calina-c to review and merge after.
I think the PR looks good, and can be merged in parallel to the remaining PRs.

Reply:
If I do the loop inside each table file then I also have to move the save function there that's why I choose to keep it inside GQL for now

Reply:
Moved logic out of GQL factory, but it's a little bit redundant now, we might want to better reuse the code once we have the Table class

Reply:
Ok, let's get the Table PR merged and then look at this.